src/main/java/seedu/address/logic/parser/SearchCommandParser.java

```diff
@@ -0,0 +1,29 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.address.commons.util.StringUtil;
+import seedu.address.logic.commands.SearchCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new SearchCommand object
+ */
+public class SearchCommandParser implements Parser<SearchCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the SearchCommand
+     * and returns a SearchCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public SearchCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, SearchCommand.MESSAGE_USAGE));
+        }
+        String moduleCode = StringUtil.ignoreCase(trimmedArgs);
+
+        return new SearchCommand(moduleCode);
```

Comment:
Maybe we can catch the possible IllegalArgumentException error thrown by the StringUtil.ignoreCase() method here so that we can display an error message to the user that something has gone wrong.

Currently if the user types in "search MA 1521", the IllegalArgumentException error is thrown in IntelliJ but not in the GUI. Maybe we can try something like that, but I'm not too sure what error message you would wanna show them hehe

```suggestion
        try {
            String moduleCode = StringUtil.ignoreCase(trimmedArgs);
            return new SearchCommand(moduleCode);
        }
        catch (IllegalArgumentException e) {
            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, SearchCommand.MESSAGE_USAGE));
        }
```

Reply:
Oh wow good catch! Added the relevant code!